src/classes/typeValidation.ts

```diff
@@ -0,0 +1,134 @@
+//this class validates the MAIN input types
+export class Validation
+{
+    constructor(){}
+
+    // ID
+    isID(id: number): boolean
+    {
+        //id &&
+        if( typeof id === 'number')
+        {
+            return true;
+        }
+        else
+        {
+            console.log("Id not valid");
+            return false;
+        }
+    }
+
+    isString(input: string): boolean
+    {
+        if(input != '' && typeof input === 'string')
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    //PHONE-NUMBER -> VARCHAR
+    isPhoneNumber(pNum: string): boolean
+    {
+        if(typeof pNum === 'string' && pNum.length > 10)
+        {
+            return true;
+        }
+        else
+        {
+            console.log("Number not valid!");
+            return false;
+        }
+    }
+
+    // NAME
+    isName(name: string): boolean
+    {
+        if(name != '' && typeof name === 'string')
+        {
+            return true;
+        }
+        else
+        {
+            console.log("Name not valid");
+            return false;
+        }
+    }
+
+    // DATE
+    isDate(date: Date): boolean
+    { 
+        //&& date.toDateString() > '01/01/1900'
+        if(date)
+        {
+           return true; 
+        }
+        else
+        {
+            console.log("Date not valid");
+            return false;
+        }
+    }
+
+    // EMAIL
+    isEmail(mail: string): boolean
+    {
+        const regexp: RegExp = new RegExp(/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);
+        return regexp.test(mail);
+    }
```

Comment:
There is a lot of debate on whether you can validate an email with regex. Use regex at your own risk